### PR TITLE
Refactor og crud

### PR DESCRIPTION
**Pull request by RaneWallin**
Still getting caught up on the crud routes. This is to get the original routes working with the new db. I have also made some improvements, including using async await and reorganizing some files to meet standard convention.

### CODE REVIEW

File: server/server.js [hunk 1]
```diff
@@ -1,12 +1,78 @@
+/* eslint-disable no-loop-func */
+/* eslint-disable no-console */
 const path = require('path');
 require('dotenv').config({
   path: path.resolve(__dirname, '../.env')
 });
-const app = require('./index.js');
+const express = require('express');
+const bodyParser = require('body-parser');
```
Comment (cywang117) [inline server/server.js:8]: Express v4 has built-in body-parser functionality via `app.use(express.json())`

File: server/server.js [hunk 2]
```diff
@@ -1,12 +1,78 @@
...
+const db = require('../db/models/index');
+const MAX_PRODUCTS = 4;
+
+const app = express();
+
+app.use(cors());
+app.use('/', express.static('public', { fallthrough: true }));
+app.use('/:product_id', express.static('public'));
+app.use(bodyParser.urlencoded({ extended: true }));
+app.use(bodyParser.json());
```
Comment (cywang117) [inline server/server.js:21]: So these two lines here can be replaced with express's built-in :)

File: server/server.js [hunk 3]
```diff
@@ -1,12 +1,78 @@
...
+
+app.get('/:product_id', (req, res) => {
+  const product_id = req.params.product_id;
+  res.redirect(`/traits/${product_id}`);
+  res.end();
+});
+
+app.get('/traits/:product_id', async (req, res) => {
+  const id = +req.params.product_id;
```
Comment (cywang117) [inline server/server.js:30]: Clever way to convert to number. Will return NaN for non-number product_id params though, might be good to factor in that edge case - maybe return early with code 400 to avoid calling db methods unnecessarily?

File: server/server.js [hunk 4]
```diff
@@ -1,12 +1,78 @@
...
+  const traits = await db.fetchTraitsForProduct(id);
```
Comment (cywang117) [inline server/server.js:31]: I noticed that your `fetchTraitsForProduct` function expression in `db/models/index.js` doesn't have a try...catch block for error handling. That's completely fine, but if that's the case, this `await` and all other `await`s below it would benefit from being wrapped in a try...catch so that the server can gracefully send back a code 500 on `fetchTraitsForProduct` error (or `await axios.get` error from line 47), instead of just crashing.

See my related comment in `db/models/index.js`

File: db/models/index.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+  results = await results.all();
+  results = getRandomGames(results);
+
+  return results;
+};
+
+const getTraitIdFromName = async (traitName) => {
+  const traitCollection = db.collection('traits');
+
+  try {
```
Comment (cywang117) [inline db/models/index.js:54]: Of the 3 function expressions in this file, this is the only one with a try...catch. Which is totally fine, of course! But your catch doesn't return out of the function.. I'm not in a position to check what would be returned from `getTraitIdFromName` when an error occurs, but I imagine that it would either return undefined, or not exit from the function at all, on account of it being a Promise? For either of those cases, it would be good practice to either handle errors here, or handle them in your `app.METHOD` lines in server.js. 

A suggestion for error handling: In `server.js`, for each endpoint:
```
app.get(ENDPOINT, async (req, res) => {
  // verify user input or parameters
  // declare globals needed in try & catch
  try {
    // all await statements here
  } catch (err) {
    console.error(err);
    res.status(500).send('An internal error occurred');
  }
});
```

And since your other two methods in `db/models/index.js` don't have try...catch blocks, `server.js` code that uses them definitely needs try...catch. Just a suggestion! If you have a better way of implementing error handling for those edge cases, by all means implement it the way you see fit.

Last note, an integration test suite for this edge case, and all other error edge cases related to using these DB methods + the server code surround them would help you catch these edge cases before they make their way to a PR 👍

**Review by cywang117 — COMMENTED** [review]
@RaneWallin Commenting without approval/requesting changes yet to get your input. What kind of error handling did you have in mind for what I mentioned?

File: db/models/index.js (hunk 5 above)
Comment (RaneWallin) [reply db/models/index.js:54]: HI Christina,

You're totally right about the lack of error handling. I was planning on adding it in when as I finish the full suite of crud functions. In this PR I was just focusing on getting the original code to work with the new database. I plan to tidy it up, get some tests written, and add error handling in the next PR.

Comment (cywang117) [reply db/models/index.js:54]: Cool cool! In that case count this as an "Approved" code review :)

Comment (RaneWallin) [reply db/models/index.js:54]: Awesome. Thanks for the feedback. I will apply it with the next PR.
